snowdrop-execution/src/Snowdrop/Execution/Expand.hs

```diff
@@ -71,21 +73,21 @@ runSeqExpandersSequentially
     )
     => Rec (ProofNExp conf) txtypes
     -> [SomeData TxRaw (Both (ExpandableTx txtypes) c)]
-    -> ERoCompM conf (UnionSeqExpandersInps txtypes) [SomeTx c]
-runSeqExpandersSequentially proofNExps allTxs =
-    reverse <$> evalStateT (foldM runExps [] allTxs) def
+    -> ERoCompM conf (UnionSeqExpandersInps txtypes) (OldestFirst [] (SomeTx c, ChgAccum conf))
+runSeqExpandersSequentially proofNExps allTxs =  do
```

Comment:
Current implementation of `runSeqExpandersSequentially` seems to be wrong:

Function `constructStateTx` relies on that `chgAccum` in the reader context is set to the state that immediately precedes transaction. But what actually happens: it remains unchanged, tx #4 would be expanded on top of state0 (not state3).